account_invoice_import_invoice2data/wizard/account_invoice_import.py

```diff
@@ -90,24 +120,89 @@ def invoice2data_parse_invoice(self, file_data):
         fileobj.close()
         return self.invoice2data_to_parsed_inv(invoice2data_res)
 
+    def invoice2data_prepare_lines(self, lines):
+        """Manipulate line data to match with account_invoice_import"""
+        for line in lines:
+            line["price_unit"] = float(line.get("price_unit", 0))
+            taxes = self.parse_invoice2data_taxes(line)
+            line["taxes"] = taxes  # or global_taxes,
+            product_dict = {
+                "barcode": line.get("barcode"),
+                "code": line.get("code", line.get("name")),
+            }
+            line["product"] = product_dict
+            if line.get("date_start"):
+                line["date_start"] = line.get("date_start")
+                line["date_end"] = line.get("date_end")
+            if line.get("line_note"):
+                line["line_note"] = line.get("line_note")
+            line["sectionheader"] = line.get("sectionheader")
+            # qty 0 should be allowed to import notes, but not supported by document_import
+            line["qty"] = float(line.get("qty", 1))
+            if line["qty"] > 0:
+                uom_dict = {
+                    "unece_code": line.get("unece_code"),
+                    "name": line.get("uom"),
+                }
+                line["uom"] = uom_dict
+                line["discount"] = float(line.get("discount", 0.0))
+                if line.get("price_subtotal"):
+                    line["price_subtotal"] = float(line.get("price_subtotal"))
+                if line.get("line_note"):
+                    line["line_note"] = line.get("line_note")
+                line["sectionheader"] = line.get("sectionheader")
+        return lines
+
     @api.model
     def invoice2data_to_parsed_inv(self, invoice2data_res):
+        lines = invoice2data_res.get("lines", [])
+        prepared_lines = self.invoice2data_prepare_lines(lines)
+
         parsed_inv = {
             "partner": {
-                "vat": invoice2data_res.get("vat"),
+                "vat": re.sub(r"\W+", "", invoice2data_res.get("vat", "")).upper(),
```

Comment:
I'd move this out to `_get_vat` or something similar. Then you can add a unit test :wink:

Reply:
Actually the convert to upper might no longer be needed.
It's already covered here.

https://github.com/OCA/edi/blob/14.0/base_business_document_import/models/business_document_import.py#L312-L313